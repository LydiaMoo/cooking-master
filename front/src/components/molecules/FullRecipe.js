import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import emptystar from '../atoms/EmptyStar.svg';

const RecipeContents = styled.article`
  display: flex; 
  flex-wrap: nowrap;
  justify-content: space-evenly; 
  align-content: center;  
  border: 2px solid #635155;
  margin: 5%; 
  box-sizing: border-box;
`;

const List = styled.ul`
  list-style-type: none;
  align-self: center; 
`;

const Image = styled.img`
  height: 200px;
  width: 200px;
  align-self: center; 
`;

const Starred = styled.img`
  height: 20px;
  align-self: center; 
  &:hover {
    background: #635155;
  }
`;

class FullRecipe extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    cookingTime: PropTypes.string.isRequired,
    ingredients: PropTypes.arrayOf(PropTypes.shape({
      quantity: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
      ]),
      name: PropTypes.string.isRequired
    })).isRequired,
    image: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired
  }

  render() {
    return (
      <RecipeContents onClick={this.props.onClick}>
          <h1>{this.props.name}</h1>
          <List>
            <li><h3>{this.props.cookingTime}</h3></li>
            {this.props.ingredients.map(({quantity, name}) => (
              <li key={name}>{quantity} of {name}</li>  
            ))}
          </List> 
        <Image src={this.props.image} alt={this.props.name}/> 
        <Starred src={emptystar} alt="starred"/>
      </RecipeContents>  
        
    );
  }
}

export default FullRecipe;