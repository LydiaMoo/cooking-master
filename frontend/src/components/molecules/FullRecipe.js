import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import StarButton from './StarButton';

const RecipeContents = styled.article`
  background: white; 
  display: flex; 
  flex-direction: column; 
  flex-wrap: nowrap;
  justify-content: space-evenly; 
  align-self: center; 
  align-items: center; 
  border: 2px solid #c9c5c6;
  border-radius: 5px; 
  padding: 0 0 5% 0 ; 
  margin: 5%; 
  width: 90%; 
  box-sizing: border-box;
`;

const Image = styled.img`
  width: 60%;
  height: auto;
  align-self: center; 
`;

const List = styled.ul`
  text-decoration: none; 
  list-style-type: none;
`;

const TitleBlock = styled.div`
  font-size: 1.5em; 
  display: flex; 
  width: 100%; 
  justify-content: space-around; 
`;

const Exit = styled.button`
  height: 20px; 
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  cursor: pointer;
  outline: inherit;
  align-self: flex-end; 
  margin: 2%; 
  &:hover{
      transform: scale(1.5);
    }
`; 

class FullRecipe extends Component {

  static defaultProps = {
		starred: false, 
	}

  static propTypes = {
    name: PropTypes.string.isRequired,
    cookingTime: PropTypes.string.isRequired,
    ingredients: PropTypes.arrayOf(PropTypes.shape({
      quantity: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
      ]),
      name: PropTypes.string.isRequired
    })).isRequired,
    image: PropTypes.string.isRequired,
    starOnChange: PropTypes.func,
    onClick: PropTypes.func,
    starred: PropTypes.bool, 
  }


  render() {
    return (
      <RecipeContents>
          <Exit onClick={this.props.onClick}><span role='img' aria-label="exit-button">‚ùå</span></Exit>
          <TitleBlock>
            <h1>{this.props.name}</h1>
            <StarButton starred={this.props.starred} starOnChange={this.props.starOnChange} />
          </TitleBlock>
          <Image src={this.props.image} alt={this.props.name}/> 
          <h3>Cooking Time: {this.props.cookingTime}</h3>
          <List><h3>To make this recipe at home, you will need:</h3> 
            {this.props.ingredients.map(({quantity, name}) => (
              <li key={name}>{quantity} {name}</li>
            ))}
          </List> 
      </RecipeContents>  
        
    );
  }
}

export default FullRecipe;