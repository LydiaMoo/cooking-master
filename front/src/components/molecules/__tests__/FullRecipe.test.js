import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FullRecipe from '../FullRecipe';
import { shallow } from 'enzyme';

const recipe = {
  name: "Lemon Cheese Cake",
  cookingTime: "40 minutes",
  ingredients: [
    {"quantity": 4, "name": "Eggs"},
    {"quantity": "1 cup", "name": "Cream Cheese"},
    {"quantity": 5, "name": "Lemon"}
  ],
  image: "https://s3-eu-west-1.amazonaws.com/s3.mediafileserver.co.uk/carnation/WebFiles/RecipeImages/lemoncheesecake_lg.jpg", 
};

describe('<FullRecipe />', () => {
  it('should match default snapshot', () => {
    const tree = renderer.create(
      <FullRecipe
        name={recipe.name} 
        cookingTime={recipe.cookingTime}
        ingredients={recipe.ingredients} 
        image={recipe.image}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('simulates click events', () => {
    const onButtonClick = jest.fn();
    const wrapper = shallow(
    <FullRecipe 
        onClick={onButtonClick}
        name={recipe.name} 
        cookingTime={recipe.cookingTime}
        ingredients={recipe.ingredients} 
        image={recipe.image}
       />);
    wrapper.find('[data-test-id="exit"]').simulate('click')
    expect(onButtonClick).toHaveBeenCalled();
  });

});
