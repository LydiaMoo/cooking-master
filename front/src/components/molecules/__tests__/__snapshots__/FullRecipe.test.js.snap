// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FullRecipe /> should match default snapshot 1`] = `
<article
  className="sc-bdVaJa flGKDD"
>
  <h1>
    Lemon Cheese Cake
  </h1>
  <ul
    className="sc-bwzfXH eBvxeR"
  >
    <li>
      <h3>
        40 minutes
      </h3>
    </li>
    <li>
      4
       of 
      Eggs
    </li>
    <li>
      1 cup
       of 
      Cream Cheese
    </li>
    <li>
      5
       of 
      Lemon
    </li>
  </ul>
  <img
    alt="Lemon Cheese Cake"
    className="sc-htpNat hYUzna"
    src="https://s3-eu-west-1.amazonaws.com/s3.mediafileserver.co.uk/carnation/WebFiles/RecipeImages/lemoncheesecake_lg.jpg"
  />
  <img
    alt="not starred"
    className="sc-bxivhb kCAVYV"
    src="EmptyStar.svg"
  />
  <button
    onClick={undefined}
  >
    <span
      aria-label="exit-button"
      role="img"
    >
      ‚ùå
    </span>
  </button>
</article>
`;
